webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js

```diff
@@ -8,42 +8,40 @@ import { translate as __ } from '../../common/I18n';
 import createModelsTableSchema from './ModelsTableSchema';
 import { getURIQuery } from '../../common/helpers';
 
-class ModelsTable extends React.Component {
-  componentDidMount() {
-    this.props.getTableItems(getURIQuery(window.location.href));
-  }
-
-  render() {
-    const {
-      getTableItems,
-      sortBy,
-      sortOrder,
-      error,
-      status,
-      results,
-    } = this.props;
+const ModelsTable = ({
+  getTableItems,
+  sortBy,
+  sortOrder,
+  error,
+  status,
+  results,
+}) => {
+  useEffect(() => {
+    getTableItems(getURIQuery(window.location.href));
+  }, []);
 
-    const renderTable =
-      status === STATUS.ERROR ? (
-        <MessageBox
-          key="models-table-error"
-          icontype="error-circle-o"
-          msg={__(`Could not receive data: ${error && error.message}`)}
-        />
-      ) : (
-        <Table
-          key="models-table"
-          columns={createModelsTableSchema(getTableItems, sortBy, sortOrder)}
-          rows={results}
-        />
-      );
-
-    if (results.length > 0) {
-      return renderTable;
-    }
+  if (results.length === 0) {
     return <Spinner size="lg" loading />;
   }
-}
+
+  if (status === STATUS.ERROR) {
```

Comment:
we should use `withRenderHandler` HOC to avoid the conditional rendering inside this component, but maybe not in this PR.

Reply:
I would prefer to keep it as is. This is going to evolve in near future i would guess, so we can keep this minimal :)